File: e2e-tests/spring-rest-mysql/src/test/java/org/evomaster/e2etests/spring/mysql/entity/SpringTestBase.java
package org.evomaster.e2etests.spring.mysql.entity;

import org.evomaster.client.java.controller.EmbeddedSutController;
import org.evomaster.e2etests.utils.RestTestBase;

public class SpringTestBase extends RestTestBase {

    protected static void initClass(EmbeddedSutController controller) throws Exception {

        RestTestBase.initClass(controller);
    }
}

File: e2e-tests/spring-rest-openapi-v3/src/test/kotlin/org/evomaster/e2etests/spring/openapi/v3/SpringTestBase.kt
package org.evomaster.e2etests.spring.openapi.v3

import org.evomaster.client.java.controller.InstrumentedSutStarter
import org.evomaster.client.java.instrumentation.InputProperties
import org.evomaster.client.java.instrumentation.InstrumentingAgent
import org.evomaster.e2etests.utils.RestTestBase
import org.junit.jupiter.api.BeforeAll

/**
 * Created by arcuri82 on 03-Mar-20.
 */
abstract class SpringTestBase : RestTestBase(){

    companion object {
        @BeforeAll
        @JvmStatic
        fun initAgent() {
            /*
                needed because kotlin.jvm.internal.Intrinsics gets loaded in
                TaintKotlinEqualController before agent is initialized
             */
            System.setProperty(InputProperties.REPLACEMENT_CATEGORIES, "BASE,SQL,EXT_0")
            InstrumentedSutStarter.loadAgent()
            InstrumentingAgent.changePackagesToInstrument("com.foo.")
        }
    }

}
